plugins/store/src/lib.rs

```diff
@@ -12,164 +12,231 @@
 )]
 
 pub use error::{Error, Result};
-use log::warn;
-use serde::Serialize;
+use serde::{Deserialize, Serialize};
 pub use serde_json::Value as JsonValue;
 use std::{
     collections::HashMap,
+    marker::PhantomData,
     path::{Path, PathBuf},
-    sync::{Mutex, Weak},
+    sync::{Arc, Mutex},
     time::Duration,
 };
-pub use store::{Store, StoreBuilder, StoreInner};
+pub use store::{DeserializeFn, SerializeFn, Store, StoreBuilder, StoreInner};
 use tauri::{
     plugin::{self, TauriPlugin},
-    AppHandle, Manager, ResourceId, RunEvent, Runtime, Webview,
+    AppHandle, Manager, ResourceId, RunEvent, Runtime, State,
 };
 
 mod error;
 mod store;
 
 #[derive(Serialize, Clone)]
+#[serde(rename_all = "camelCase")]
 struct ChangePayload<'a> {
     path: &'a Path,
```

Comment:
Not sure if we still want to keep this, this used to be the way to identify a store from the js side